legend-engine-ide-lsp-extension-api/src/main/java/org/finos/legend/engine/ide/lsp/extension/text/TextLocation.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2024 Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.finos.legend.engine.ide.lsp.extension.text;
+
+import java.util.Objects;
+
+/**
+ * A text source location, including the source URI and the text interval on that source URI
+ */
+public class TextLocation implements Comparable<TextLocation>
+{
+    private final String sourceUri;
+    private final TextInterval textInterval;
+
+    private TextLocation(String sourceUri, TextInterval textInterval)
+    {
+        this.sourceUri = Objects.requireNonNull(sourceUri, "Source URI is required");
+        this.textInterval = Objects.requireNonNull(textInterval, "Text interval is required");
+    }
+
+    /**
+     * The source URI for this source
+     * @return source URI
+     */
+    public String getSourceUri()
+    {
+        return sourceUri;
+    }
```

Comment:
Should this be a source URI or just a generic source id? If we're going to call it source URI, should we enforce that it's a valid URI?

Reply:
We could, but on the eclipse LPS library, source URI's are just represented as String without any further validation.  I'm not sure  about their rational for it, but I think we should be consistent with this.  

I can rename it to source id, or something else if you think make the code more readable.